internal/manifests/internal/config/options.go

```diff
@@ -44,9 +47,22 @@ type Parallelism struct {
 	QueryFrontendReplicas int32
 }
 
+// WriteAheadLog for ingester processing
+type WriteAheadLog struct {
+	Directory             string
+	IngesterMemoryRequest int64
+}
+
 // Value calculates the floor of the division of
 // querier cpu limits to the query frontend replicas
 // available.
 func (p Parallelism) Value() int32 {
 	return int32(math.Floor(float64(p.QuerierCPULimits) / float64(p.QueryFrontendReplicas)))
 }
+
+// ReplayMemoryCeiling calculates 75% of the ingester memory
+// for the ingester to use for the write-ahead-log capbability.
+func (w WriteAheadLog) ReplayMemoryCeiling() string {
+	value := int64(math.Ceil(float64(w.IngesterMemoryRequest) * float64(0.75)))
+	return fmt.Sprintf("%d", value)
+}
```

Comment:
According to the official Loki production configuration (at least this is the public version of what Grafana Cloud is using), they recomend 50% (https://github.com/grafana/loki/blob/main/production/ksonnet/loki/wal.libsonnet#L17). Could you elaborate on why 75%?

Reply:
Hmm, there appears to be a discrepancy here. The documentation recommended 75% (https://grafana.com/docs/loki/latest/operations/storage/wal/).